plugin/core/sessions.py

```diff
@@ -1622,6 +1643,12 @@ def m_workspace_semanticTokens_refresh(self, params: Any, request_id: Any) -> No
             else:
                 sv.session_buffer.set_semantic_tokens_pending_refresh()
 
+    def m_workspace_inlayHint_refresh(self, params: None, request_id: Any) -> None:
+        """handles the workspace/inlayHint/refresh request"""
+        for sv in self.session_views_async():
```

Comment:
Instead of requesting new inlay hints for all of the views, maybe it would be better to use the logic that I used at the semantic tokens refresh. Meaning that only requests for the views that are currently visible ( = selected) are made, and for the other views a variable is set that a request should be made as soon as the view gets activated. See also the description of the refresh requests in the lsp specs, that say clients can delay request for non-visible tabs. The reason I did it this way was because one server (I think it was rust-analyzer) would send a refresh request after each text change. And if someone has for example 10 tabs in the window, but only one of it active, then we send 9 useless requests all the time and do the inlay hints processing and phantom updates even for views that are not visible currently.

Reply:
I do not think that I would work on optimizing this.

Inlay hints doesn't seem like an expensive request (like semantic token is). 
If this turns to be a problem only than I would optimize and add more code.

Here is one implementation detail of how inlay hints currently function.
If the server implements the workspace/inlayHint/refresh request,
we will request inlay hints for each session view.

So the client will refresh the inlay hints each time the server desires,
as an end result we should not end up with stale inlay hints in one view ever.

Even if the server supports or does not support the workspace/inlayHint/refresh request,
there is this peace of code that will request inlay hints as soon as the view gets active:

https://github.com/sublimelsp/LSP/pull/2018/files#diff-982c110867d9809c978a5597e4da66ce7538b327675a66b8f27214413b7c0ec1R345

This logic is redundant for servers that do support workspace/inlayHint/refresh request,
but I would say that it is necessary for servers for that do not support workspace/inlayHint/refresh. (just in case to update stale inlay hints as soon as the user focuses one view)

But as I said I would not add any code for optimizing this until it turns out to be a perf issue.